data-access-layer/iaas/AzureClient.js

```diff
@@ -27,6 +27,58 @@ class AzureClient extends BaseCloudClient {
       .value()
     );
     Promise.promisifyAll(this.computeClient.snapshots);
+    Promise.promisifyAll(this.computeClient.disks);
+  }
+
+  getSnapshot(snapshotId) {
+    return this.computeClient.snapshots.getAsync(this.settings.resource_group, snapshotId)
+      .then(snapshotResponse => _.pick(snapshotResponse, ['location', 'creationData', 'id']));
+  }
+
+  createDiskFromSnapshot(snapshotId, zones, opts = {}) {
+    const diskName = this.getRandomDiskId();
+    return Promise.try(() => this.getSnapshot(snapshotId))
+      .then(snapshotData => {
+        const options = {
+          zones: _.isArray(zones) ? zones : [zones],
+          location: snapshotData.location,
+          tags: _.assign({}, opts.tags || {}, {
+            createdBy: 'service-fabrik'
+          }),
+          sku: _.assign({}, opts.sku || {}, {
+            name: 'Premium_LRS',
+            tier: 'Premium'
+          }),
+          creationData: {
+            createOption: 'Copy',
+            sourceUri: snapshotData.id
+          }
+        };
+        return this.computeClient.disks.createOrUpdateAsync(this.settings.resource_group, diskName, options);
+      })
+      .then(diskResponse => ({
+        volumeId: diskResponse.name,
+        size: diskResponse.diskSizeGB,
+        zone: diskResponse.zones[0],
+        type: diskResponse.sku ? diskResponse.sku.name : '',
+        extra: {
+          sku: diskResponse.sku
+        }
```

Comment:
As I commented above, I thought the `extra` field would contain tags and any iaas-specific properties in the response

Reply:
@abh1kg sure, understood. I guess it is not yet used, but maybe used later. But since sku, is already part of type, do we need it here? And maybe a better name would be good. :)